app/src/main/java/com/example/tasky/agenda/di/AgendaModule.kt

```diff
@@ -0,0 +1,35 @@
+package com.example.tasky.agenda.di
+
+import com.example.tasky.agenda.data.DateGeneratorImpl
+import com.example.tasky.agenda.data.remote.AgendaRepositoryImpl
+import com.example.tasky.agenda.data.remote.AgendaService
+import com.example.tasky.agenda.domain.AgendaRepository
+import com.example.tasky.agenda.domain.DateGenerator
+import dagger.Binds
+import dagger.Module
+import dagger.Provides
+import dagger.hilt.InstallIn
+import dagger.hilt.components.SingletonComponent
+import retrofit2.Retrofit
+
+
+@Module
+@InstallIn(SingletonComponent::class)
+class AgendaModule {
+
+    @Provides
```

Comment:
Remember the @Singleton

Reply:
I disagree here, the rule that I follow for Singleton is asking if that class must preserve some data if not it must be factory for not having that class use memory or resources when is not being used and since a Retrofit service doesnt need to preserve some data I prefer to use the default Factory

Reply:
What Singleton does, is reuse the same instance over and over, as its scoped to the lifecycle of the app (@ViewModelScope is scoped to the lifecycle of a viewmodel, @Service to the service, etc)

So by not having Singleton, the app will create a new instance everytime the instance is being injected